provider/cldr/src/transform/locale_canonicalizer/likelysubtags.rs

```diff
@@ -2,6 +2,7 @@
 // called LICENSE at the top level of the ICU4X source tree
```

Comment:
Should this file be called `likely_subtags.rs`? It's not one word, and there's `provider/cldr/src/cldr_serde/likely_subtags.rs`.

Reply:
+1

We have some inconsistency here. I would prefer to see them all with an underscore if possible. 
![image](https://user-images.githubusercontent.com/58569820/143083639-9c9ae443-6be5-4673-ba18-65d386cdbe74.png)

Reply:
The `likelysubtags@1.json` file is based on the data key.  That is an internal thing that shouldn't generally be used by humans.  The field is also limited to 16 characters; although the underscore would technically fit, I'd like to see those strings remain narrow.

I'm happy to rename the `.rs` files to have an underscore though.